### PR TITLE
feat(pt_expt): add training infrastructure

### LINKED ISSUE #5225: PyTorch Exportable Training Loop Implementation
Implement the training loop for the PyTorch exportable backend.

**Goals:**
- Define the training step, loss calculation, and optimizer integration compatible with export workflows.
- Ensure parity with the TensorFlow training workflow.

---
*Generated by OpenClaw*

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
## Summary by CodeRabbit

* **New Features**
  * New "PyTorch-Exportable" backend with end-to-end training CLI, model factory, loss wrapper, model wrapper, checkpointing, neighbor/stat utilities, and optional torch.compile acceleration with automatic re-tracing.
* **Bug Fixes**
  * Added runtime validation for required parameter fields (fparam/aparam) and ensured loss/reporting arrays are device-consistent.
* **Refactor**
  * Reworked stat input, batch normalization, and model-stat collection for clearer, backend-agnostic data flow.

### CODE REVIEW

**Review by njzjz — APPROVED** [review]
LGTM

File: deepmd/pt_expt/entrypoints/main.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            "Calculate neighbor statistics... "
+            "(add --skip-neighbor-stat to skip this step)"
+        )
+        type_map = config["model"].get("type_map")
+        train_data = get_data(config["training"]["training_data"], 0, type_map, None)
+        from deepmd.pt_expt.model import (
+            BaseModel,
+        )
+
+        config["model"], _min_nbor_dist = BaseModel.update_sel(
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply deepmd/pt_expt/entrypoints/main.py:152]: fixed by [f0f260d](https://github.com/deepmodeling/deepmd-kit/pull/5270/commits/f0f260d873698a9cc5232221b25b8de402ab2504)

**Review by iProzd — APPROVED** [review]
